src/mds/MDLog.cc

```diff
@@ -736,12 +738,14 @@ void MDLog::trim()
       break;
     }
 
+    auto interval = std::chrono::duration<double>(last_trim - trim_start);
+    auto should_trim = is_oversegmented() && (interval.count() >= oversegmented_idle_interval);
```

Comment:
Should the timestamp difference expression be `trim_start - last_trim` instead ?
I think the expression `last_trim - trim_start` might turn out negative.

Reply:
That should only be valid for the first iteration.